### DISCUSSION TITLE
Usage of OnNavigatedTo  from inside a module

### DISCUSSION THREAD

**Discussion by danielmbomfim** [microsoft/react-native-windows#10539] [Q&A] [answered]
Hi,

I'm trying to create a native module for my C++/WinRT app in react-native. The objective is to create a native implementation of the google oauth2 flow. I am using this sample as a base https://github.com/googlesamples/oauth-apps-for-windows/tree/master/OAuthUniversalApp

In the sample the authentication is done on the browser and the  OnNavigatedTo method of the app is overwriten to allow the app to get the authorization code from the callback url

there's a way of overwrite this method from inside the module or at least to subscribe to an event to get this code?

Thanks

**Comment by chrisglein** [comment, marked as answer]
The modules don't have a built in API to interact with the Application object, or the Page, or the Window. Which I think is what you're looking for here. This would be something you could accomplish in an app, but not obviously today from the module without some creative thinking.

**Comment by danielmbomfim** [comment]
> The modules don't have a built in API to interact with the Application object, or the Page, or the Window. Which I think is what you're looking for here. This would be something you could accomplish in an app, but not obviously today from the module without some creative thinking.

There's a way I could call the native module from the app class? That way I could include a installation manual step where the developer overrides the app onNavigatedTo method and from there call the module method to handle the response from the auth page
